packages/mattermost-redux/src/reducers/entities/users.ts

```diff
@@ -0,0 +1,638 @@
+        return profileListToSet(state, action);
+
+    case UserTypes.RECEIVED_PROFILES_LIST_NOT_IN_TEAM_AND_REPLACE:
+        return profileListToSet(state, action, true);
+
+    case UserTypes.RECEIVED_PROFILE_IN_TEAM:
+        return removeProfileFromSet(state, action);
+
+    case UserTypes.RECEIVED_PROFILES_LIST_IN_TEAM:
+        return removeProfileListFromSet(state, action);
+
+    case UserTypes.LOGOUT_SUCCESS:
+        return {};
+
+    case UserTypes.PROFILE_NO_LONGER_VISIBLE:
+        return removeProfileFromTeams(state, action);
+
+    default:
+        return state;
+    }
+}
+
+function profilesWithoutTeam(state: Set<string> = new Set(), action: GenericAction) {
+    switch (action.type) {
+    case UserTypes.RECEIVED_PROFILE_WITHOUT_TEAM: {
+        const nextSet = new Set(state);
+        Object.values(action.data as string[]).forEach((id: string) => nextSet.add(id));
+        return nextSet;
+    }
+    case UserTypes.RECEIVED_PROFILES_LIST_WITHOUT_TEAM: {
+        const nextSet = new Set(state);
+        action.data.forEach((user: UserProfile) => nextSet.add(user.id));
+        return nextSet;
+    }
+    case UserTypes.PROFILE_NO_LONGER_VISIBLE:
+    case UserTypes.RECEIVED_PROFILE_IN_TEAM: {
+        const nextSet = new Set(state);
+        nextSet.delete(action.data.id);
+        return nextSet;
+    }
+    case UserTypes.LOGOUT_SUCCESS:
+        return new Set();
+
+    default:
+        return state;
+    }
+}
+
+function profilesInChannel(state: RelationOneToMany<Channel, UserProfile> = {}, action: GenericAction) {
+    switch (action.type) {
+    case UserTypes.RECEIVED_PROFILE_IN_CHANNEL:
+        return addProfileToSet(state, action);
+
+    case UserTypes.RECEIVED_PROFILES_LIST_IN_CHANNEL:
+        return profileListToSet(state, action);
+
+    case UserTypes.RECEIVED_PROFILES_IN_CHANNEL:
+        return profilesToSet(state, action);
+
+    case UserTypes.RECEIVED_PROFILE_NOT_IN_CHANNEL:
+        return removeProfileFromSet(state, action);
+
+    case ChannelTypes.CHANNEL_MEMBER_REMOVED:
+        return removeProfileFromSet(state, {
+            type: '',
+            data: {
+                id: action.data.channel_id,
+                user_id: action.data.user_id,
+            }});
+
+    case UserTypes.LOGOUT_SUCCESS:
+        return {};
+
+    case UserTypes.PROFILE_NO_LONGER_VISIBLE:
+        return removeProfileFromTeams(state, action);
+
+    case UserTypes.RECEIVED_BATCHED_PROFILES_IN_CHANNEL: { // Used by the mobile  app
+        const {data} = action;
+
+        if (data && data.length) {
+            const nextState = {...state};
+            data.forEach((d: any) => {
+                const {channelId, users} = d.data;
+                const nextSet = new Set(state[channelId]);
+
+                users.forEach((u: UserProfile) => nextSet.add(u.id));
+                nextState[channelId] = nextSet as any; // TODO the type of state should Record<string, Set<string>>
```

Comment:
This one was a weird one. The type of `nextSet` seems to be incorrect since everywhere else expects each entry in `state` to be an array instead of a Set. Assuming the above comment isn't lying, this was only used by the mobile app, so I didn't bother fixing it now.